File: JS/tests/search_insert_position.test.js
import { test, expect } from 'vitest'
import { searchInsert } from '../src/search_insert_position.js'

const testCases = [
  { nums: [1, 3, 5, 6], target: 5, expected: 2 },
  { nums: [1, 3, 5, 6], target: 2, expected: 1 },
  { nums: [1, 3, 5, 6], target: 7, expected: 4 },
  { nums: [1, 3, 5, 6], target: 0, expected: 0 },
  { nums: [1], target: 0, expected: 0 },
  { nums: [1], target: 2, expected: 1 },
  { nums: [2, 4, 6, 8, 10], target: 7, expected: 3 },
  { nums: [2, 4, 6, 8, 10], target: 11, expected: 5 }
]

testCases.forEach(({ nums, target, expected }) => {
  test(`should return ${expected} for input [${nums}] and target ${target}`, () => {
    expect(searchInsert(nums, target)).toBe(expected)
  })
})

File: C++/tests/search_insert_position_test.cc
#include "../src/search_insert_position.hpp"
#include "gtest/gtest.h"
#include <gtest/gtest.h>
#include <tuple>
#include <vector>

class SearchInsertPositionTest
    : public testing::TestWithParam<tuple<vector<int>, int, int>> {
protected:
  Solution solution;
};

TEST_P(SearchInsertPositionTest, ) {
  auto [nums, target, expected] = GetParam();

  int actual = solution.searchInsert(nums, target);

  EXPECT_EQ(actual, expected);
}

INSTANTIATE_TEST_SUITE_P(
    , SearchInsertPositionTest,
    testing::Values(make_tuple(vector<int>{1, 3, 5, 6}, 5, 2),
                    make_tuple(vector<int>{1, 3, 5, 6}, 2, 1),
                    make_tuple(vector<int>{1, 3, 5, 6}, 7, 4),
                    make_tuple(vector<int>{1, 2, 3, 4}, 5, 4),
                    make_tuple(vector<int>{1, 2, 3}, 0, 0),
                    make_tuple(vector<int>{1, 3}, 2, 1)));
